File: Common/Actor/src/common/actor/TextureBuffer.java
package common.actor;

import javafx.scene.image.Image;

import java.io.IOException;
import java.util.HashMap;

final class TextureBuffer {

    private static final HashMap<String, Image> fileToImage = new HashMap<>();

    static Image loadImage( final String fileName ) {
        if ( !fileToImage.containsKey( fileName ) ) {
            try {
                fileToImage.put(fileName, ImageLoader.loadImage(fileName ));
            } catch (NullPointerException | IOException npe ) {
                npe.printStackTrace();
            }
        }
        return fileToImage.get( fileName );
    }
}

File: Common/Actor/src/common/actor/TextureBuffer.kt
package common.actor

import javafx.scene.image.Image

internal object TextureBuffer {
    private val fileToImage = hashMapOf<String, Image>()

    infix fun loadImage(fileName: String): Image {
        if (!fileToImage.containsKey(fileName)) {
            fileToImage[fileName] = ImageLoader.loadImage(fileName)
        }
        return fileToImage[fileName]!!
    }
}
